core/src/mindustry/editor/MapEditorDialog.java

```diff
@@ -666,18 +667,34 @@ private void tryExit(){
     }
 
     private void addBlockSelection(Table table){
-        Table content = new Table();
-        pane = new ScrollPane(content);
+        blockSelection = new Table();
+        pane = new ScrollPane(blockSelection);
         pane.setFadeScrollBars(false);
         pane.setOverscroll(true, false);
         pane.exited(() -> {
             if(pane.hasScroll()){
                 Core.scene.setScrollFocus(view);
             }
         });
-        ButtonGroup<ImageButton> group = new ButtonGroup<>();
 
-        int i = 0;
+        Table searchBar = new Table();
+        searchBar.image(Icon.zoom);
+        searchBar.field("", this::rebuildBlockSelection)
+        .name("editor/search").maxTextLength(maxNameLength).get().setMessageText("@players.search");
+
+        table.add(searchBar).pad(10);
```

Comment:
Should be defined with `.table(t -> {/** add field, image, etc*/})`

Also, 10 is too much padding. Try something like 4-5.

Reply:
https://github.com/Anuken/Mindustry/pull/3265/commits/275395b5536586a90ce0b79fe20ce90f198963b9